### DISCUSSION TITLE
Bypassing aws IAM user access key

### DISCUSSION THREAD

**Discussion by cjabbour** [justeat/JustSaying#1068] [Q&A]
Can this library be used without specifying a specific IAM user access and secret access key? Our DevOps engineers would prefer to add a role to our resources like EC2 to allow them the appropriate permissions rather than using IAM users' access keys in the application.

**Comment by martincostello** [comment]
Yes it can. That's how we use JustSaying in our AWS-hosted applications (QA, staging, production).

We typically keep usage with keys for engineers' local development.

Ultimately, we rely on the default behaviours of the AWS SDKs themselves and their built-in strategies for resolving credentials (which you can override and be explicit with if you desire).
